src/python/turicreate/toolkits/_model.py

```diff
@@ -514,17 +514,17 @@ def summary(self, output=None):
         >>> m.summary()
         """
         if output is None or output == 'stdout':
-            pass
+            try:
+                print(self.__repr__())
+            except:
+                return self.__class__.__name__
         elif (output == 'str'):
             return self.__repr__()
         elif output == 'dict':
-            return _toolkit_serialize_summary_struct( self, \
+            return _toolkit_serialize_summary_struct(self, \
                                             *self._get_summary_struct() )
-
-        try:
-            print(self.__repr__())
-        except:
-            return self.__class__.__name__
+        else:
+            raise ToolkitError("Unsupported argument " + str(output) + " for summary.")
```

Comment:
For error message, I like to make it very clear it's referring to a parameter. I would change: `summary.` to `\"summary\" parameter.`

Reply:
good!